src/input/InputComponent.cpp

```diff
@@ -132,7 +132,7 @@ void InputComponent::handleAction(const QString& action)
         else
         {
           qDebug() << "Invoking slot" << qPrintable(recvSlot->m_slot.data());
-          auto arg0 = QMetaMethodArgument();
+          QGenericArgument arg0 = QGenericArgument();
```

Comment:
This change causes my build to fail on Arch Linux:

```
[proc] Executing command: /usr/bin/cmake --build /home/zjeffer/Documents/repos/jellyfin-media-player/build --config Debug --target all --
[build] [1/4  25% :: 0.052] Automatic MOC for target JellyfinMediaPlayer
[build] [2/3  66% :: 2.415] Building CXX object src/CMakeFiles/JellyfinMediaPlayer.dir/input/InputComponent.cpp.o
[build] FAILED: src/CMakeFiles/JellyfinMediaPlayer.dir/input/InputComponent.cpp.o 
[build] /usr/bin/g++ -DHAVE_CEC -DHAVE_LIRC -DHAVE_MINIZIP -DHAVE_SDL -DLINUX_DBUS=1 -DPREFIX=\"/usr/local\" -DQS_LOG_LINE_NUMBERS -DQS_LOG_SEPARATE_THREAD -DQT_CORE5COMPAT_LIB -DQT_CORE_LIB -DQT_DBUS_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_OPENGL_LIB -DQT_POSITIONING_LIB -DQT_QMLINTEGRATION_LIB -DQT_QMLMODELS_LIB -DQT_QML_LIB -DQT_QUICK_LIB -DQT_WEBCHANNELQUICK_LIB -DQT_WEBCHANNEL_LIB -DQT_WEBENGINECORE_LIB -DQT_WEBENGINEQUICK_LIB -DQT_WIDGETS_LIB -DQT_XML_LIB -DUSE_X11EXTRAS -DUSE_X11XRANDR -I/home/zjeffer/Documents/repos/jellyfin-media-player/build/src -I/home/zjeffer/Documents/repos/jellyfin-media-player/src -I/home/zjeffer/Documents/repos/jellyfin-media-player/build/src/JellyfinMediaPlayer_autogen/include -I/include -I/usr/include/SDL2 -I/home/zjeffer/Documents/repos/jellyfin-media-player/X11_X11_INCLUDE_PATH -I/home/zjeffer/Documents/repos/jellyfin-media-player/X11_Xrandr_INCLUDE_PATH -I/home/zjeffer/Documents/repos/jellyfin-media-player/external/SPMediaKeyTap -I/home/zjeffer/Documents/repos/jellyfin-media-player/external/HIDRemote -I/home/zjeffer/Documents/repos/jellyfin-media-player/external/plistparser -I/home/zjeffer/Documents/repos/jellyfin-media-player/external/letsmove -I/home/zjeffer/Documents/repos/jellyfin-media-player/src/player -I/home/zjeffer/Documents/repos/jellyfin-media-player/src/core -I/home/zjeffer/Documents/repos/jellyfin-media-player/src/shared -isystem /usr/include/qt6/QtCore -isystem /usr/include/qt6 -isystem /usr/lib/qt6/mkspecs/linux-g++ -isystem /usr/include/qt6/QtCore5Compat -isystem /usr/include/qt6/QtNetwork -isystem /usr/include/qt6/QtQml -isystem /usr/include/qt6/QtQmlIntegration -isystem /usr/include/qt6/QtWebChannel -isystem /usr/include/qt6/QtGui -isystem /usr/include/qt6/QtGui/6.6.1 -isystem /usr/include/qt6/QtGui/6.6.1/QtGui -isystem /usr/include/qt6/QtCore/6.6.1 -isystem /usr/include/qt6/QtCore/6.6.1/QtCore -isystem /usr/include/qt6/QtQuick -isystem /usr/include/qt6/QtQmlModels -isystem /usr/include/qt6/QtOpenGL -isystem /usr/include/qt6/QtWidgets -isystem /usr/include/qt6/QtWebEngineQuick -isystem /usr/include/qt6/QtWebEngineCore -isystem /usr/include/qt6/QtPositioning -isystem /usr/include/qt6/QtWebChannelQuick -isystem /usr/include/qt6/QtXml -isystem /usr/include/qt6/QtDBus -Wall -fno-omit-frame-pointer -fvisibility-inlines-hidden  -g -std=gnu++17 -mno-direct-extern-access -MD -MT src/CMakeFiles/JellyfinMediaPlayer.dir/input/InputComponent.cpp.o -MF src/CMakeFiles/JellyfinMediaPlayer.dir/input/InputComponent.cpp.o.d -o src/CMakeFiles/JellyfinMediaPlayer.dir/input/InputComponent.cpp.o -c /home/zjeffer/Documents/repos/jellyfin-media-player/src/input/InputComponent.cpp
[build] In file included from /usr/include/qt6/QtCore/qmetatype.h:18,
[build]                  from /usr/include/qt6/QtCore/qsharedpointer_impl.h:30,
[build]                  from /usr/include/qt6/QtCore/qsharedpointer.h:12,
[build]                  from /usr/include/qt6/QtCore/qdebug.h:16,
[build]                  from /usr/include/qt6/QtCore/QDebug:1,
[build]                  from /home/zjeffer/Documents/repos/jellyfin-media-player/src/input/InputComponent.cpp:1:
[build] /home/zjeffer/Documents/repos/jellyfin-media-player/src/input/InputComponent.cpp: In member function ‘void InputComponent::handleAction(const QString&)’:
[build] /home/zjeffer/Documents/repos/jellyfin-media-player/src/input/InputComponent.cpp:138:20: error: no match for ‘operator=’ (operand types are ‘QGenericArgument’ and ‘QMetaMethodArgument’)
[build]   138 |             arg0 = Q_ARG(const QString&, hostArguments);
[build]       |                    ^~~~~
[build] /usr/include/qt6/QtCore/qobjectdefs.h:82:21: note: candidate: ‘constexpr QGenericArgument& QGenericArgument::operator=(const QGenericArgument&)’
[build]    82 | class Q_CORE_EXPORT QGenericArgument
[build]       |                     ^~~~~~~~~~~~~~~~
[build] /usr/include/qt6/QtCore/qobjectdefs.h:82:21: note:   no known conversion for argument 1 from ‘QMetaMethodArgument’ to ‘const QGenericArgument&’
[build] /usr/include/qt6/QtCore/qobjectdefs.h:82:21: note: candidate: ‘constexpr QGenericArgument& QGenericArgument::operator=(QGenericArgument&&)’
[build] /usr/include/qt6/QtCore/qobjectdefs.h:82:21: note:   no known conversion for argument 1 from ‘QMetaMethodArgument’ to ‘QGenericArgument&&’
[build] ninja: build stopped: subcommand failed.
[proc] The command: /usr/bin/cmake --build /home/zjeffer/Documents/repos/jellyfin-media-player/build --config Debug --target all -- exited with code: 1
```

Using Qt 6.6.1. 

@satmandu  Which version of Qt did you use for this to build properly?

Reply:
I built with whatever came with Ubuntu 23.0.4...

Reply:
Looks like Ubuntu 23.04 uses Qt 6.4.2, according to [the documentation](https://doc.qt.io/qt-6.5/qmetaobject.html#Q_ARG) the Q_ARG has [a different return value](https://doc.qt.io/qt-6.2/qmetaobject.html#Q_ARG).

I'll create a PR to make this work for all Qt versions.